api/types/kustomization.go

```diff
@@ -298,6 +292,39 @@ func (k *Kustomization) FixKustomizationPreMarshalling(fSys filesys.FileSystem)
 	return nil
 }
 
+// fixPatchesStrategicMerge converts PatchesStrategicMerge to Patches.
+func (k *Kustomization) fixPatchesStrategicMerge(fSys filesys.FileSystem) error {
+	if k.PatchesStrategicMerge == nil {
+		return nil
+	}
+
+	for _, patchStrategicMerge := range k.PatchesStrategicMerge {
+		// check this patch is file path select.
+		if _, err := fSys.ReadFile(string(patchStrategicMerge)); err == nil {
```

Comment:
Nit: the first thing `splitPatchStrategicMerge` below needs to do is repeat this read. Could we keep the value and pass it in?

Reply:
Good point.  I actually think attempting to Read the file isn't a good way to determine if this is a file patch or an inline patch.

I'm thinking something like `if fSys.Exists(string(patchStrategicMerge)) {` might be more appropriate, avoids the double-read, and also matches the intention.